shared_model/packages/javascript/.gitignore

```diff
@@ -0,0 +1,67 @@
+# Logs
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# Runtime data
+pids
+*.pid
+*.seed
+*.pid.lock
+
+# Directory for instrumented libs generated by jscoverage/JSCover
+lib-cov
+
+# Coverage directory used by tools like istanbul
+coverage
+
+# nyc test coverage
+.nyc_output
+
+# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
+.grunt
+
+# Bower dependency directory (https://bower.io/)
+bower_components
+
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (http://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
+jspm_packages/
+
+# Typescript v1 declaration files
+typings/
+
+# Optional npm cache directory
+.npm
+
+# Optional eslint cache
+.eslintcache
+
+# Optional REPL history
+.node_repl_history
+
+# Output of 'npm pack'
+*.tgz
+
+# Yarn Integrity file
+.yarn-integrity
+
+# dotenv environment variables file
+.env
+
+# Build and generated directories
+build/*
+pb/*
+lib/*
+
+package-lock.json
```

Comment:
Why do we gitignore package-lock? I always thought what we should commit it to repo
[commit it to repo](https://docs.npmjs.com/files/package-lock.json) so our users have exact same dependencies.

Reply:
It's generally a bad idea to use `.lock` files with native NPM packages, because it may cause problems with the local building same project in different environments.

But `.lock` still be the nice feature for JS-only or prebuilt projects. We should investigate the correct usage of it in the case of native packages.